### PR TITLE
Code switcher implementation

### PR DESCRIPTION
**Pull request by gorzelinski**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Docs: Code switcher implementation

---

### Additional information

This PR also contains other small improvements like:

- formatting
- typos
- grammar
- snippets

### CODE REVIEW

File: docs/getting-started/setup/getting-and-setting-data.md
Comment (gorzelinski) [inline docs/getting-started/setup/getting-and-setting-data.md:1]: In this guide, we promote import maps in one snippet (Automatic integration with HTML forms section). I'm guessing we want to rewrite it using globals?

File: docs/getting-started/setup/getting-and-setting-data.md [hunk 1]
```diff
@@ -219,8 +219,8 @@ The example below shows how all these mechanisms can be used together to enable
 </info-box>
 -->
 
+<code-switcher>
 ```js
-// Note: We need to build the editor from source.
```
Comment (gorzelinski) [inline docs/getting-started/setup/getting-and-setting-data.md:223]: It was probably a left-over comment from the previous installation methods.

**Comment by gorzelinski** [comment]
We also promote import maps in the UI language guide (setup). Should we rewrite it as well?

Screenshot 2024-10-04 at 12 35 53

File: docs/getting-started/setup/getting-and-setting-data.md
Comment (Witoso) [reply docs/getting-started/setup/getting-and-setting-data.md:1]: yes, let's use globals.

**Comment by Witoso** [comment]
> We also promote import maps in the UI language guide (setup). Should we rewrite it as well?

Yep, good find!

**Comment by godai78** [comment]
An infobox that they work differently because it's an external code.

**Comment by gorzelinski** [comment]
There is also the Mermaid experimental plugin. It is also not imported from the main CKEditor 5 package (based on the text in the guide). So, I guess that the plugin is not accessible from the cloud.

**Comment by godai78** [comment]
Not now, but wasn't it planned to be made a real feature at some point?

**Comment by godai78** [comment]
We lost the icons:

**Comment by gorzelinski** [comment]
> We also promote import maps in the UI language guide (setup). Should we rewrite it as well?

> Yep, good find!

Update: Filip had a good idea. We could show all installation methods but in a particular order, like:

1. npm
2. CDN + UMD
3. CDN + import maps

**Comment by Witoso** [comment]
Let's discuss it, but 3 options is a lot, and I'm worried people won't understand the difference. Import maps have a very low understanding, and in this code switcher we don't have a lot of place to explain it.

**Comment by godai78** [comment]
We could link it to the getting started section. But that would actually make a mess in the long run, probably. However, while it's not very understandable, it would be nice to have it clearly explained.

**Comment by gorzelinski** [comment]
> We lost the icons:

I can't reproduce it, so the best I can do is to say, "Works for me." But I know it's not helpful, so I'd be grateful if someone could build docs on this branch.

**Comment by gorzelinski** [comment]
To summarize (potential) follow-ups:

1. Decide what to do with external plugins (MathType, WProofreader)
2. Decide what to do with our experimental plugin - Mermaid
3. Discuss import maps in the UI language guide

**Review by godai78 — APPROVED** [review]

File: packages/ckeditor5-clipboard/docs/features/drag-drop.md [hunk 2]
```diff
@@ -72,6 +73,7 @@ ClassicEditor
 	.then( /* ... */ )
 	.catch( /* ... */ );
 ```
+</code-switcher>
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/docs/features/drag-drop.md:76]: There's still the `// Or 'GPL'.` comment

File: packages/ckeditor5-clipboard/docs/features/paste-plain-text.md [hunk 3]
```diff
@@ -50,6 +51,7 @@ ClassicEditor
 	.then( /* ... */ )
 	.catch( /* ... */ );
 ```
+</code-switcher>
```
Comment (filipsobol) [inline packages/ckeditor5-clipboard/docs/features/paste-plain-text.md:54]: There's still the `// Or 'GPL'.` comment

File: packages/ckeditor5-code-block/docs/features/code-blocks.md [hunk 4]
```diff
@@ -46,6 +47,7 @@ ClassicEditor
 	.then( /* ... */ )
 	.catch( /* ... */ );
 ```
+</code-switcher>
```
Comment (filipsobol) [inline packages/ckeditor5-code-block/docs/features/code-blocks.md:50]: There's still the `// Or 'GPL'.` comment

File: packages/ckeditor5-find-and-replace/docs/features/find-and-replace.md [hunk 5]
```diff
@@ -44,6 +45,7 @@ ClassicEditor
 	.then( /* ... */ )
 	.catch( /* ... */ );
 ```
+</code-switcher>
```
Comment (filipsobol) [inline packages/ckeditor5-find-and-replace/docs/features/find-and-replace.md:48]: There's still the `// Or 'GPL'.` comment

**Comment by Witoso** [comment]
Please don't wait for my review, I'm deep into other topics. Will review everything :grimacing:

**Review by filipsobol — APPROVED** [review]

**Pull request merged**